Tests/MathLib/TestEigenCSR.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+
+#include <vector>
+#include <numeric>
+
+#include <Eigen/Sparse>
+
+
+/**
+ * This test case checks if the internal Eigen::SparseMatrix compressed storage format
+ * is a conventional CSR matrix. Currently this is the case, but it is not guaranteed
+ * for all time.
+ *
+ * Cf. section "Sparse matrix format" on page
+ * http://eigen.tuxfamily.org/dox/group__TutorialSparse.html
+ */
+TEST(MathLibEigen, Eigen2CSR)
+{
+    const int nrows = 16;
+    const int ncols = nrows;
+
+    Eigen::SparseMatrix<double, Eigen::RowMajor> mat(nrows, ncols);
+
+    // set up sparsity pattern
+    std::vector<int> pat(nrows);
+    for (std::size_t i=0; i<nrows; ++i)
+    {
+        if (i==0 || i==nrows-1) {
+            pat[i] = 2;
+        } else {
+            pat[i] = 3;
+        }
+    }
+
+    // CSR representation of the matrix
+    std::vector<double> values;
+    std::vector<int> ia(nrows+1); // row offsets
+    std::vector<int> ja;          // column indices
+
+    std::partial_sum(pat.begin(), pat.end(), ia.begin()+1);
+
+    const int nnz = ia.back();
+    values.reserve(nnz);
+    ja.reserve(nnz);
+
+    mat.reserve(pat);
+
+    // init matrix, build CSR matrix in parallel
+    for (int row=0; row<nrows; ++row) {
+        for (int col = -1; col<=1; ++col) {
+            int cidx = row + col;
+            if (cidx < 0 || cidx >= ncols) continue;
+
+            const double val = (col == 0) ? 2.0 : -1.0;
+            values.push_back(val);
+            mat.coeffRef(row, cidx) = val;
+            ja.push_back(cidx);
+        }
+    }
+
+    // change matrix
```

Comment:
Personally, I would remove the loops (line 73 - 83) and lines 84, 85. In case you want to keep them I would remove at least the outermost ("repeat") loop and modify the matrix entries in the same manner than the values in the vector `values` (line 85) are modified, i.e., `mat.coeffRef(row, cidx) *= 4.0;`